---
float64[] grip_loc
